### PR TITLE
adding support for `filterModel`

### PR DESCRIPTION
**Pull request by BSd3v**
- info from the grid (#109)
- added output for `test_filter`
- added check for action props to delay until the grid is ready (#99)
- adjusted and fixed html templates (#116)

### CODE REVIEW

File: src/lib/fragments/AgGrid.react.js [hunk 1]
```diff
@@ -329,16 +329,14 @@ export default class DashAgGrid extends Component {
...
             this.state.gridApi.forEachNodeAfterFilter((node) => {
                 virtualRowData.push(node.data);
             });
-
-            const filterModel = this.state.gridApi.getFilterModel();
-            this.setState({filterModel});
-            setProps({virtualRowData});
         }
+        const filterModel = this.state.gridApi.getFilterModel();
+        setProps({virtualRowData, filterModel});
```
Comment (alexcjohnson) [inline src/lib/fragments/AgGrid.react.js:339]: Do we actually want to be setting `virtualRowData` if `rowModelType` is not `'clientSide'`? If not, we can make a `const propsToSet = {filterModel}` and inside the `if` we set `propsToSet.virtualRowData = virtualRowData`.

Comment (BSd3v) [reply src/lib/fragments/AgGrid.react.js:339]: Ok, yes. That's a better idea. 

I was avoiding the setProps for the virtualRowData by itself for obvious reasons, but also, it was clearing the filters too.

File: src/lib/fragments/AgGrid.react.js [hunk 2]
```diff
@@ -528,6 +539,43 @@ export default class DashAgGrid extends Component {
...
+            this.deleteSelectedRows(false);
+        }
+
+        setProps({
+            exportDataAsCsv: false,
+            selectAll: false,
+            deselectAll: false,
+            autoSizeAllColumns: false,
+            deleteSelectedRows: false,
+        });
```
Comment (alexcjohnson) [inline src/lib/fragments/AgGrid.react.js:577]: Also this one - we should only `setProps` the ones that weren't already `false`.

Comment (BSd3v) [reply src/lib/fragments/AgGrid.react.js:577]: Alright, I'm using the same way the above works. `propsToSet` then build through each, then if `propsToSet` not empty, it `setProps`

File: CHANGELOG.md [hunk 3]
```diff
@@ -76,6 +76,9 @@ Links "DE#nnn" prior to version 2.0 point to the Dash Enterprise closed-source D
   - allowed for functions, styleConditions and regular dictionaries to be passed to the `cellStyle` on all levels
   - added `rowId` to `cellClicked` data
 
+- [#111](https://github.com/plotly/dash-ag-grid/pull/111)
+  - fixing templates to only populate when `dangerously_allow_code=True`
```
Comment (alexcjohnson) [inline CHANGELOG.md:80]: That note belongs under `Fixed` - but we should also note the addition of `filterModel` in the `Added` section

File: src/lib/fragments/AgGrid.react.js [hunk 4]
```diff
@@ -528,6 +548,47 @@ export default class DashAgGrid extends Component {
...
+
+        if (deleteSelectedRows) {
+            this.deleteSelectedRows(false);
+            propsToSet.deleteSelectedRows = false;
+        }
+
+        if (!isEmpty(propsToSet)) {
+            setProps({
+                ...propsToSet,
+            });
```
Comment (alexcjohnson) [inline src/lib/fragments/AgGrid.react.js:589]:
```suggestion
            setProps(propsToSet);
```

**Review by alexcjohnson — APPROVED** [review]
Excellent! 💃 pending changelog edit.

File: CHANGELOG.md (hunk 3 above)
Comment (BSd3v) [reply CHANGELOG.md]: adjusted via https://github.com/plotly/dash-ag-grid/pull/111/commits/679ed74b1af6dbeff7c142d63c62c7f845e893c7

**Pull request merged**